backend/src/resources/todos/todo.router.ts

```diff
@@ -1,16 +1,31 @@
-import { NextFunction, Request, Response, Router } from 'express';
+import {  Request, Response, Router } from 'express';
 import { Todo } from './todo.model';
 
-const todos: Todo[] = [];
+let todos: Todo[] = [];
 const todosRouter = Router();
 
-todosRouter.get('/', (req: Request, res: Response, next: NextFunction) => {
-  res.json({
-    message: 'Todo: Implement',
-  });
+todosRouter.get('/', (req: Request, res: Response) => {
+  res.json(todos);
 });
 
-todosRouter.post('/', (req: Request, res: Response, next: NextFunction) => {
+todosRouter.delete('/:id', (req: Request, res: Response) => {
+  if (todos.findIndex(todo => todo.id === req.params.id) !== -1) {
+    todos = todos.filter(todo => todo.id !== req.params.id);
+    res.json({ status: true });
+  } else {
+    res.json({ status: false });
+  }
+});
+
+todosRouter.patch('/:id', (req: Request, res: Response) => {
+  const index: number = todos.findIndex(todo => todo.id === req.params.id);
+  if (index !== -1) {
+    todos[index].isDone = req.body.isDone;
+    res.json(todos[index]);
+  }
+});
+
+todosRouter.post('/', (req: Request, res: Response) => {
   const todo: Todo = {
     id: Math.random().toString(36).substring(2, 7),
     description: req.body.description,
```

Comment:
Add a check to ensure `description` is present in the request payload before creating a Todo.